retrace/metric_backend_opengl.cpp

```diff
@@ -307,15 +311,20 @@ void MetricBackend_opengl::beginPass() {
 }
 
 void MetricBackend_opengl::processQueries() {
-    int64_t gpuStart, gpuEnd, pixels;
+    int64_t gpuStart, gpuEnd, timestamp, pixels;
     for (int i = 0; i < QUERY_BOUNDARY_LIST_END; i++) {
         QueryBoundary boundary = static_cast<QueryBoundary>(i);
         while (!queries[i].empty()) {
             auto &query = queries[i].front();
             if (metrics[METRIC_GPU_START].profiled[i]) {
                 glGetQueryObjecti64v(query[QUERY_GPU_START], GL_QUERY_RESULT,
-                                     &gpuStart);
-                int64_t value = gpuStart - baseTime;
+                                      &timestamp);
+                if (timestamp < gpuStart) {
+                    // Correct clock domain of baseTime after roll-over
+                    baseTime -= pow(2, timestamp_bits_precision);
```

Comment:
These are all integer quantities, whereas `pow` is a function of doubles.

Let's use `UINT64_C(2) << timestamp_bits_precision` instead, here and below.